scripts/freshvenvs.py

```diff
@@ -124,6 +155,28 @@ def _versions_fully_cover_bounds(bounds: typing.Tuple[str, str], versions: typin
     return versions[0] >= Version(upper_bound)
 
 
+def _iterate_venvs(venv: riotfile.Venv, package: str) -> bool:
+    """
+    DFS through the Venv, as it may have nested Venvs.
+    Returns whether the package was pinned or not.
+
+    If package is in module_suite_mapping, remap it.
+    """
+    package = module_suite_mapping.get(package, package)
+
+    if package in venv.pkgs:
+        if "" in venv.pkgs[package]:
+            return False
+
+    if venv.venvs:
+        for child_venv in venv.venvs:
+            is_pinned = _iterate_venvs(child_venv, package)
```

Comment:
"pinned" is a tricky concept in this context, because there are lots of patterns we can use in the riotfile other than `latest` that specify a range of versions rather than one specific version (examples: "~=1.2.3", ">=2.3.4"). You could sidestep that trickiness by renaming `_iterate_venvs` to `_venv_sets_latest_for_package` or similar, and inverting the return values.

Reply:
Thank you for the catch and calling that out!